### PR TITLE
chore(baseid): add index

### PR DESCRIPTION
**Pull request by philloooo**
also refactor the code to not do 2 lookups for baseid and did.

### CODE REVIEW

File: indexd/index/drivers/alchemy.py [hunk 1]
```diff
@@ -509,18 +509,13 @@ def get(self, did):
         """
         with self.session as session:
             query = session.query(IndexRecord)
-            query = query.filter(IndexRecord.did == did)
+            query = query.filter(
+                or_(IndexRecord.did == did, IndexRecord.baseid == did)
+                ).order_by(IndexRecord.created_date.desc())
```
Comment (phuongph) [inline indexd/index/drivers/alchemy.py]: Can you let a user flag here?
`def get(self, did, flag=True):`

`if flag:`
       `query = query.filter(or_ (...))`
`else:`
       `query = query.filter(IndexRecord.did == did)`

File: indexd/index/drivers/alchemy.py [hunk 2]
```diff
@@ -905,9 +902,18 @@ def migrate_7(session, **kwargs):
...
 
+def migrate_8(session, **kwargs):
+    """
+    create index on IndexRecord.baseid
+    """
+    session.execute(
+        "CREATE INDEX ix_{tb}_baseid ON {tb} ( baseid )"
+        .format(tb=IndexRecord.__tablename__))
+
+
```
Comment (phuongph) [inline indexd/index/drivers/alchemy.py]: Should check if `ix_index_record_baseid` index exist before create a new one

File: indexd/index/drivers/alchemy.py (hunk 1 above)
Comment (phuongph) [reply indexd/index/drivers/alchemy.py]: This can be very useful for my services

Comment (philloooo) [reply indexd/index/drivers/alchemy.py]: I don't think we should keep adding flags for different tweaks unless it's necessary. We can merge this and if the performance is not good, we can think about adding flags later

File: indexd/index/drivers/alchemy.py (hunk 2 above)
Comment (philloooo) [reply indexd/index/drivers/alchemy.py]: None of the existing migration check that because it's relying on the migration version table

**Pull request merged**